python/ray/serve/_private/application_state.py

```diff
@@ -572,7 +573,18 @@ def _reconcile_target_deployments(self) -> None:
 
         # Set target state for each deployment
         for deployment_name, info in self._target_state.deployment_infos.items():
-            self.apply_deployment_info(deployment_name, info)
+            deploy_info = deepcopy(info)
```

Comment:
I think the logic itself looks good, just flagging that we will execute a deepcopy once per deployment for every control loop iteration. I think it would be more efficient to store a copy of this info object since it doesn't change between control loop iterations.